src/server/src/proxy.ts

```diff
@@ -761,14 +742,14 @@ async function processRequest(query, req, res) {
   }
 
   // Block non-allowed RPC commands
-  if (!query.action || user_allowed_commands.indexOf(query.action) === -1) {
+  if (!query.action || user_allowed_commands?.indexOf(query.action) === -1) {
```

Comment:
`?` is optional chaining (used multiple places here). Since we don't know if `user_allowed_commands` actually is set. We could avoid these cases by making `user_allowed_commands` and similar variables always be an empty array.